explorer/data/prelude.carbon

```diff
@@ -107,3 +107,37 @@ class Heap {
 }
 
 var heap: Heap = {};
+
+interface Eq {
```

Comment:
Please can you define an `EqWith` following https://github.com/carbon-language/carbon-lang/blob/trunk/docs/design/expressions/comparison_operators.md#equality ? We don't support `constraint` yet, so I don't think we can define `Eq` in terms of it, but this should avoid the need to change the `==` implementation when that works.